roles/users/tasks/main.yml

```diff
@@ -50,7 +50,7 @@
   changed_when: false
   sudo: no
   register: admin_user_status
-  when: not sshd_permit_root_login
+  when: not sshd_permit_root_login and ansible_ssh_host is defined
```

Comment:
Very strange. It seems like an entry in your Ansible `hosts` file implies "connect to this host via SSH," which Ansible does no matter what you pass with `--connection`. And it sets `ansible_connection` accordingly, whether or not it is actually able to connect. It does this even for inventories passed as string with `-i` (e.g., `-i "[web] 127.0.0.1,"`—that comma is important). This is especially strange because this is how Packer does it.

If, however, you set the host to `127.0.0.1` in the playbook via `hosts: 127.0.0.1`, it works. I'll take another look this weekend.